packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -23,6 +23,15 @@
 using namespace realm::jni_util;
 using namespace realm::_impl;
 
+jobject wrap_long(JNIEnv* jenv, jlong long_value, jboolean managed = false) {
```

Comment:
I think you could avoid having to wrap longs if you use a functional interface instead of a lambda on the RealmInterop compact method. It would allow you to define primitive types arguments instead of being forced to use boxed types.